chain-impl-mockchain/src/certificate/vote_tally.rs

```diff
@@ -56,9 +66,28 @@ impl VoteTally {
         self.payload.payload_type()
     }
 
+    pub fn decryption_shares(&self) -> Option<Box<[Box<[TallyDecryptShare]>]>> {
```

Comment:
I think there should be a newtype like said above, this return type signature is painful to read and it's unclear what the array dimensions mean.

Reply:
Something like this:

```rust
pub struct VoteDecryptedShares {
    inner: Box<[Box<[TallyDecryptShare]>],
}

impl VoteDecryptedShares {
    pub fn proposal_shares(&self, proposal_id: u8) -> &[TallyDecryptShare] {
        &self.inner[proposal_id as usize]
    }
}
```